subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java

```diff
@@ -1558,10 +1558,16 @@ public void addRuleBasedPluginListener(RuleBasedPluginListener listener) {
     public void prepareForRuleBasedPlugins() {
         if (!preparedForRuleBasedPlugins) {
             preparedForRuleBasedPlugins = true;
+            addModelAndComponentReportingTasks();
```

Comment:
Instead of applying plugins here, can we just apply the plugins in `ComponentBasePlugin`?  This should be used by all software model based plugins.

Reply:
Heh, that's where I originally had it, but it felt wrong, so I moved it to here.  We have a bunch of tests that were applying rule-based plugins without applying component-base and I found myself adding these plugins everywhere.  I don't feel very strongly about it one way or the other - it just felt like all the changes required were a signal that it wasn't the best approach.

Reply:
I've updated it to apply via `ComponentBasePlugin`.